tests/tests.rs

```diff
@@ -47,6 +47,134 @@ fn test_valid_private_keys() {
+        &Strkey::HashTx(StrkeyHashTx([
+            0x36, 0x3e, 0xaa, 0x38, 0x67, 0x84, 0x1f, 0xba, 0xd0, 0xf4, 0xed, 0x88, 0xc7, 0x79,
+            0xe4, 0xfe, 0x66, 0xe5, 0x6a, 0x24, 0x70, 0xdc, 0x98, 0xc0, 0xec, 0x9c, 0x07, 0x3d,
+            0x05, 0xc7, 0xb1, 0x03,
+        ])),
+    );
+}
+
+#[test]
+fn test_valid_hash_x() {
+    assert_convert_roundtrip(
+        "XA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQH5FV",
+        &Strkey::HashX(StrkeyHashX([
+            0x36, 0x3e, 0xaa, 0x38, 0x67, 0x84, 0x1f, 0xba, 0xd0, 0xf4, 0xed, 0x88, 0xc7, 0x79,
+            0xe4, 0xfe, 0x66, 0xe5, 0x6a, 0x24, 0x70, 0xdc, 0x98, 0xc0, 0xec, 0x9c, 0x07, 0x3d,
+            0x05, 0xc7, 0xb1, 0x03,
+        ])),
+    );
+}
+
+#[test]
+fn test_valid_muxed_ed25519() {
+    // Valid multiplexed account (id: 123456)
+    assert_convert_roundtrip(
+        "MA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQGAAAAAAAAAPCICBKU",
+        &Strkey::MuxedAccountEd25519(StrkeyMuxedAccountEd25519([
+            0x36, 0x3e, 0xaa, 0x38, 0x67, 0x84, 0x1f, 0xba, 0xd0, 0xf4, 0xed, 0x88, 0xc7, 0x79,
+            0xe4, 0xfe, 0x66, 0xe5, 0x6a, 0x24, 0x70, 0xdc, 0x98, 0xc0, 0xec, 0x9c, 0x07, 0x3d,
+            0x05, 0xc7, 0xb1, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xe2, 0x40,
+        ])),
+    );
+
+    // Valid multiplexed account (id: 0)
+    assert_convert_roundtrip(
+        "MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAACJUQ",
+        &Strkey::MuxedAccountEd25519(StrkeyMuxedAccountEd25519([
+            0x3f, 0x0c, 0x34, 0xbf, 0x93, 0xad, 0x0d, 0x99, 0x71, 0xd0, 0x4c, 0xcc, 0x90, 0xf7,
+            0x05, 0x51, 0x1c, 0x83, 0x8a, 0xad, 0x97, 0x34, 0xa4, 0xa2, 0xfb, 0x0d, 0x7a, 0x03,
+            0xfc, 0x7f, 0xe8, 0x9a, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        ])),
+    );
+
+    // Valid multiplexed account in which unsigned id exceeds maximum signed 64-bit integer
+    assert_convert_roundtrip(
+        "MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVAAAAAAAAAAAAAJLK",
+        &Strkey::MuxedAccountEd25519(StrkeyMuxedAccountEd25519([
+            0x3f, 0x0c, 0x34, 0xbf, 0x93, 0xad, 0x0d, 0x99, 0x71, 0xd0, 0x4c, 0xcc, 0x90, 0xf7,
+            0x05, 0x51, 0x1c, 0x83, 0x8a, 0xad, 0x97, 0x34, 0xa4, 0xa2, 0xfb, 0x0d, 0x7a, 0x03,
+            0xfc, 0x7f, 0xe8, 0x9a, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        ])),
+    );
+}
+
+#[test]
+fn test_invalid_muxed_ed25519() {
+    // TODO: This test case is supposed to fail, but it will pass, I think this is the responsibility of the base32 lib
+    // maybe related to https://github.com/stellar/rs-stellar-strkey/issues/10
+    // The unused trailing bit must be zero in the encoding of the last three bytes (24 bits) as five base-32 symbols (25 bits)
+    // let r = Strkey::from_string("MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAACJUR");
+    // assert_eq!(r, Err(DecodeError::Invalid));
```

Comment:
Good catch, this is a problem. I wasn't convinced the base32 lib is the right one to be using when I started this crate. We can address this issue separately to this PR.